### PR TITLE
Noble forwarding wasm helpers

### PR DESCRIPTION
**Pull request by grod220**
Related to: https://github.com/prax-wallet/prax/issues/206 & https://github.com/penumbra-zone/penumbra/pull/4878

Exposes helpers needed to generate noble forwarding addresess.

### CODE REVIEW

**Review by JasonMHasperhoven — APPROVED** [review]

**Comment by JasonMHasperhoven** [comment]
Recommend that someone else looks at this before merging since I'm not too familiar with rust.

**Review by vacekj — APPROVED** [review]
LGTM.

File: packages/wasm/crate/tests/test_keys.rs [hunk 1]
```diff
@@ -87,3 +87,39 @@ fn returns_false_on_unknown_addr() {
             .0;
     assert!(!is_controlled_address(&fvk.encode_to_vec(), &other_address.encode_to_vec()).unwrap());
 }
+
+#[test]
```
Comment (TalDerei) [inline packages/wasm/crate/tests/test_keys.rs:91]: what about additional test coverage for:

1. edge cases near the sequence range boundaries (eg. check addresses are different for sequence numbers 0 and 2^16 - 1)

2. same sequence and account generate the same address

3. different full viewing keys yield different addresses

**Review by TalDerei — APPROVED** [review]
wasm achieves parity with core implementation

File: packages/wasm/crate/tests/test_keys.rs (hunk 1 above)
Comment (grod220) [reply packages/wasm/crate/tests/test_keys.rs:91]: Adding those additional cases 👍
